modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ues/ui/integration/test/dashboard/BannerTest.java

```diff
@@ -0,0 +1,384 @@
+import org.wso2.carbon.automation.engine.context.beans.User;
+import org.wso2.carbon.integration.common.admin.client.UserManagementClient;
+import org.wso2.ues.integration.common.clients.ResourceAdminServiceClient;
+import org.wso2.ues.ui.integration.util.UESUIIntegrationTest;
+import org.wso2.ues.ui.integration.util.UESWebDriver;
+import ues.integration.tests.common.domain.UESIntegrationTestConstants;
+
+import javax.xml.xpath.XPathExpressionException;
+import java.io.File;
+import java.net.MalformedURLException;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Created by lasanthas on 10/8/15.
+ */
+public class BannerTest extends UESUIIntegrationTest {
+
+    private static final Log LOG = LogFactory.getLog(BannerTest.class);
+
+    private static final String DASHBOARD_TITLE = "Banner Dashboard";
+    private static final String DASHBOARD_ID = "banner-dashboard";
+    private static final String ROOT_RESOURCE_PATH = "/_system/config/ues/customizations/";
+    private static final String[] IMAGES = {"orange.png", "silver.png"};
+
+    private static final String EDITOR_ROLE = "editor";
+    private static final String VIEWER_ROLE = "viewer";
+    private static final String ADMIN_ROLE = "admin";
+
+    private ResourceAdminServiceClient resourceAdminServiceClient;
+    private User editor, viewer;
+
+    /**
+     * Initialize the class
+     */
+    public BannerTest() throws Exception {
+        super(TestUserMode.SUPER_TENANT_ADMIN);
+
+        AutomationContext automationContext = new AutomationContext(UESIntegrationTestConstants
+                .UES_PRODUCT_NAME, this.userMode);
+
+        editor = automationContext.getContextTenant().getTenantUser("editor");
+        viewer = automationContext.getContextTenant().getTenantUser("viewer");
+
+        resourceAdminServiceClient = new ResourceAdminServiceClient(getBackEndUrl(),
+                getCurrentUsername(), getCurrentPassword());
+
+        UserManagementClient userManagementClient = new UserManagementClient(getBackEndUrl(),
+                getCurrentUsername(), getCurrentPassword());
+
+        // Create user roles
+        String[] editorsRoles = new String[]{editor.getUserName()};
+        String[] viewersRoles = new String[]{viewer.getUserName()};
+
+        userManagementClient.addRole(EDITOR_ROLE, editorsRoles, null);
+        userManagementClient.addRole(VIEWER_ROLE, viewersRoles, null);
+
+        // Remove unnecessary roles from users
+        String[] rolesToRemove = new String[]{ADMIN_ROLE};
+
+        userManagementClient.addRemoveRolesOfUser(editor.getUserName(), null, rolesToRemove);
+        userManagementClient.addRemoveRolesOfUser(viewer.getUserName(), null, rolesToRemove);
+    }
+
+    /**
+     * Setup the testing environment
+     *
+     * @throws Exception
+     */
+    @BeforeClass(alwaysRun = true)
+    public void setUp() throws Exception {
```

Comment:
Check the possibility of throwing the specific exception rather than throwing Exception class